### PR TITLE
bpo-43795: Add a test for Stable ABI symbol availability using ctypes

### PR DESCRIPTION
**Pull request by encukou**
This is a cross-platform check that the symbols are actually
exported in the ABI, not e.g. hidden in a macro.

https://bugs.python.org/issue43795

### CODE REVIEW

File: Tools/scripts/stable_abi.py [hunk 1]
```diff
@@ -259,6 +259,37 @@ def gen_doc_annotations(manifest, args, outfile):
             'added': item.added,
             'ifdef_note': ifdef_note})
 
+@generator("ctypes_test", 'Lib/test/test_stable_abi_ctypes.py')
+def gen_ctypes_test(manifest, args, outfile):
+    """Generate/check the ctypes-based test for exported symbols"""
+    write = partial(print, file=outfile)
+    write(textwrap.dedent('''
+        # Generated by Tools/scripts/stable_abi.py
```
Comment (erlend-aasland) [inline Tools/scripts/stable_abi.py:266]: Should this be checked by the _"Check if generated files are up to date"_ CI?

Comment (encukou) [reply Tools/scripts/stable_abi.py:266]: Yes. I made the classic mistake, renaming without running the tests :/

**Comment by encukou** [comment]
Aha, I didn't realize `ctypes` is an optional module. Thanks, *Gentoo Installed with X PR* buildbot!

**Comment by encukou** [comment]
So, still not quite there:
- *AMD64 Arch Linux TraceRefs PR* reports missing symbols `PyModule_Create2` and `PyModule_FromDefAndSpec2`!
- *s390x Fedora Refleaks PR* and *x86 Gentoo Installed with X PR* unrelated (`test_asyncio` timeout, `test_idle` failure)

**Comment by vstinner** [comment]
> test_asyncio timeout

I guess that it's https://bugs.python.org/issue44112 Yesterday, I increased the buildbot timeout from 3 hours to 4 hours to be able to debug this issue.

> test_idle failure

I should be fixed: https://bugs.python.org/issue44282

**Comment by vstinner** [comment]
> AMD64 Arch Linux TraceRefs PR reports missing symbols PyModule_Create2 and PyModule_FromDefAndSpec2!

I guess that it's a similar issue than https://bugs.python.org/issue44133

We should try to link the Python binary from objects files (.o) rather than using the libpython static library.

**Pull request merged**